lib/plug/upload.ex

```diff
@@ -60,28 +60,86 @@ defmodule Plug.Upload do
     end
   end
 
+  @doc """
+  Assign ownership of the given upload file to another process.
+
+  Useful if you want to do some work on an uploaded file in another process
+  since it means that the file will survive the end of the request.
+  """
+  @spec give_away(t | binary, pid, pid) :: :ok | {:error, binary}
+  def give_away(upload, to_pid, from_pid \\ self())
+
+  def give_away(%__MODULE__{path: path}, to_pid, from_pid) do
+    give_away(path, to_pid, from_pid)
+  end
+
+  def give_away(path, to_pid, from_pid)
+      when is_binary(path) and is_pid(to_pid) and is_pid(from_pid) do
+    with [{^from_pid, _tmp}] <- :ets.lookup(@dir_table, from_pid),
+         true <- is_path_owner?(from_pid, path) do
+      case :ets.lookup(@dir_table, to_pid) do
+        [{^to_pid, _tmp}] ->
+          :ets.insert(@path_table, {to_pid, path})
+          :ets.delete_object(@path_table, {from_pid, path})
+
+          :ok
+
+        [] ->
+          server = plug_server()
+
+          with {:ok, tmps} <- GenServer.call(server, :roots),
+               {:ok, tmp} <- generate_tmp_dir(tmps),
+               :ok <- GenServer.call(server, {:give_away, to_pid, tmp, path}) do
+            :ets.delete_object(@path_table, {from_pid, path})
+            :ok
+          else
+            {:no_tmp, _tmps} ->
```

Comment:
I think we should either return this to the caller OR simply assume it cannot fail by matching on `{:ok, tmp} = generate_tmp_dir(tmps)`. I am more inclined to the latter. Thoughts?

Reply:
The latter, agreed - it really is unlikely